rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlVisitor.java

```diff
@@ -163,9 +165,18 @@ private boolean keyMatches(Yaml.Mapping m1, Yaml.Mapping m2) {
     }
 
     private Mapping mergeMapping(Yaml.Mapping m1, Yaml.Mapping m2, P p, Cursor cursor) {
+        // Merge same key, different value together
         List<Yaml.Mapping.Entry> mergedEntries = map(m1.getEntries(), existingEntry -> {
-            for (Entry incomingEntry : m2.getEntries()) {
+            for (Yaml.Mapping.Entry incomingEntry : m2.getEntries()) {
                 if (keyMatches(existingEntry, incomingEntry)) {
+                    // multiline scalar cannot be formatted automatically, apply indent from existing value
+                    if (shouldAutoFormat && incomingEntry.getValue() instanceof Yaml.Scalar && hasLineBreak(((Scalar) incomingEntry.getValue()).getValue())) {
+                        String s = grabAfterFirstLineBreak(((Yaml.Scalar) existingEntry.getValue()).getValue());
```

Comment:
This solution is not very nice imho. It should 'just work' like the `add` stuff, by adding the `MultilineScalarAdded` marker, so the IndentsVisitor can pick it up.

At the moment the IndentsVisitor does not put the right idents in front, I'll will look into this next Monday.